chore: add hatch (#6442)
DataDog/dd-trace-py: docs/contributing-testing.rst

How do I run the test suite?
----------------------------

We assume you have `docker <https://www.docker.com/products/docker>`_ installed.

In addition, you will need `riot <https://ddriot.readthedocs.io/en/latest/>`_ and `hatch <https://hatch.pypa.io/latest/>`_.

.. code-block:: bash

    $ pip install riot==0.17.4
    $ pip install hatch==1.7.0

> Our test environments are managed with riot. However, we are migrating to hatch.

Reviewer: ```suggestion
Some of our test environments are managed with Riot, others with Hatch.
```
This wording is timeless and won't need to be updated for quite a while.

For riot environments, you can run:

.. code-block:: bash

    $ scripts/ddtest riot run -p 3.10

This command runs the entire test suite, which is probably not what you want to do.

> For hatch environments, you can run:

Reviewer: This isn't in scope for this pull request, but the two paths in this documentation suggest that it could be helpful to provide a script that makes the hatch/riot decision so contributors don't have to.
